core/src/main/java/org/infinispan/persistence/PersistenceUtil.java

```diff
@@ -154,21 +154,21 @@ public static String sanitizedCacheName(String cacheName) {
       return cacheName.replaceAll("[^a-zA-Z0-9-_\\.]", "_");
    }
 
-   public static Path getQualifiedLocation(GlobalConfiguration globalConfiguration, String location, String cacheName, String qualifier) {
+   public static Path getQualifiedLocation(GlobalConfiguration globalConfiguration, String location, String cacheName, String... qualifiers) {
```

Comment:
I know this will make the code a bit uglier, but I wonder if we should change the signature to `getQualifiedLocation(GlobalConfiguration globalConfiguration, String location, String cacheName, String qualifier, String... additionalQualifiers)`.

Or it seems the qualifiers are optional now?

Reply:
I've added a new commit that actually removes my changes to this class. I was thinking about it in the wrong way, instead we should add the "data" directory like other stores as it's more descriptive.